### PR TITLE
Add chat related functionality

### LINKED ISSUE #239: System message in chat service when changing question or language
chat message to include "<user> have change question/language from ... to ..." when question/language is changed

### PR DESCRIPTION
**Pull request by JrmCkh**
This PR consists of adding Chat related functionalities:

- Broadcasting to chat whenever a user changes a question/language
- Auto-scroll to bottom of chat, whenever a new message is received
- Enable 'enter' button to send message instead of solely via 'send button'

Closes #239

### CODE REVIEW

File: frontend/src/components/Collaboration/Chat.js [hunk 1]
```diff
@@ -38,13 +41,24 @@ const Chat = ({ socket, roomId, user }) => {
...
           <div
             key={index}
-            className={`message ${msg.sender === user ? 'self' : 'other'}`}
+            className={`message ${
+              msg.sender === user
+                ? 'self'
+                : msg.sender === 'server'
+                ? 'server'
+                : 'other'
+            }`}
```
Comment (lhy-hoyin) [inline frontend/src/components/Collaboration/Chat.js:61]: IMO, a bit hard to read. Does this looks better?
```js
className={`message ${
  msg.sender === user
  ? 'self'
  : (msg.sender === 'server' ? 'server' : 'other')
}`}
```

File: frontend/src/components/Collaboration/Chat.js [hunk 2]
```diff
@@ -53,15 +67,18 @@ const Chat = ({ socket, roomId, user }) => {
...
-        <button onClick={handleSendMessage}>Send</button>
-      </div>
+      <form onSubmit={handleSendMessage}>
+        <div className='chat-input'>
+          <input
+            type='text'
+            placeholder='Enter a message'
+            value={inputMessage}
+            onChange={handleInputChange}
+            onKe
```
Comment (lhy-hoyin) [inline frontend/src/components/Collaboration/Chat.js:77]: Is this code `onKe` useful?

**Review by lhy-hoyin — CHANGES_REQUESTED** [review]

File: frontend/src/components/Collaboration/Chat.js (hunk 1 above)
Comment (JrmCkh) [reply frontend/src/components/Collaboration/Chat.js]: Changed

File: frontend/src/components/Collaboration/Chat.js (hunk 2 above)
Comment (JrmCkh) [reply frontend/src/components/Collaboration/Chat.js]: Rmoved

**Review by lhy-hoyin — APPROVED** [review]
LGTM

**Pull request merged**